pyphare/pyphare/pharesee/run.py

```diff
@@ -89,6 +89,42 @@ def _compute_current(patch):
                 {"name":"Jy", "data":Jy, "centering":centering["Jy"]},
                 {"name":"Jz", "data":Jz,"centering":centering["Jz"]})
 
+
+def _divB2D(Bx,By, xBx, yBy):
+    dxbx = (Bx[1:,:]-Bx[:-1,:])/(xBx[1]-xBx[0])
+    dyby = (By[:,1:]-By[:,:-1])/(yBy[1]-yBy[0])
+    return dxbx + dyby
+
+def _compute_divB(patch):
+
+    if patch.box.ndim ==1:
+        raise ValueError("divB is 0 by construction in 1D")
+
+    elif patch.box.ndim == 2:
+        By  = patch.patch_datas["By"].dataset[:]
+        Bx  = patch.patch_datas["Bx"].dataset[:]
+        xBx = patch.patch_datas["Bx"].x
+        yBy = patch.patch_datas["By"].y
+        divB = _divB2D(Bx,By, xBx, yBy)
+
+        return ({"name":"cell_custom", "data":divB, "centering":"dual"},)
+
+    else:
+        raise RuntimeError("dimension not implemented")
+
+
+
+def _get_rank(patch):
+    if patch.box.ndim == 2:
+        shape = patch.box.shape
+        ghost_shape = (shape[0]+5, shape[1]+5)
```

Comment:
I know we made this choice that we have 5 ghost points... but it would be nicer to write this value somewhere (close to refinement_ratio=2 for example) so that it appears more explicit

Reply:
ok TODO, I did this because passing the shape as is didn't work but I have not thought that it was not +5-5 anymore so why does it work as is.... to investigate

Reply:
ok this is now fixed. The issue was that FieldData.x (or y, z) calls layout.yeeCoordsFor() which assumes withGhosts=True so not making a dataset that has the shape of the ghost box results in a shape mismatch with the coordinates when calling pcolormesh in plot2d.